### PR TITLE
531 532 Add details of FDC3 versioning and deprecation policies to website

### LINKED ISSUE #531: Identifying Standard Versions
## Formalizing the FDC3 Standard documentation

This issue has been raised as part of work, sponsored by FINOS, to improve and formalize the documentation of the FDC3 standard.

##  Issue description 

Typically, a Standard that has marketplace relevance is revised from time to time, to correct errors and/or to add functionality to support new use cases.

The FDC3 Standard uses a major and minor version number, as in 1.0, 1.1, 1.2. … 2.0, … *m*.*n*.

It is useful in a standard to say what is new/different in the current version as compared to the previous one. And it likely is useful to retain a list of differences between older versions.

**Proposal(s):**

1. Include the version number in the Standard’s title.
1. At the very beginning of “Getting Started” say something like the following:
> This Standard defines FDC3 Version *m*.*n*. The differences between this version and earlier ones can be found in [link to change log](xx).

### LINKED ISSUE #532: Feature Deprecation
## Formalizing the FDC3 Standard documentation

This issue has been raised as part of work, sponsored by FINOS, to improve and formalize the documentation of the FDC3 standard.

##  Issue description 

Over time, it is not uncommon for certain things in a standard to be marked for removal in a future version, possibly being replaced by an alternative.  That is, they are *deprecated*. Often, they are retained in the standard because of their widespread use, but their use in new projects is discouraged.

> It is worth noting that a given deprecated feature might *never* actually be removed, and even if it is, implementations may well continue to support it as an extension! (In fact, when some standards committees have removed features, they have been subject to major objections from the public, and, in rare cases, even legal challenges!)

The FDC3 Standard applies deprecation in two ways: In API|Spec, it has a number of *Deprecation notice*s, and in API|References, it has a number of `/** @deprecated */` JSDoc comments in TypeScript code.

As a professional courtesy, a Standard’s committee typically gives at least one version’s-worth of advance warning *before* actually removing a feature or changing some syntax. This gives the implementer and user community time to consider and retool for the implications of such a change. That said, the committee could decide to make such a change late in the process of producing a new version, *without* giving any such warning. However, such a last-minute decision may well prove to be cause for regret later on.

How should a change in behavior of an existing feature be handled? The feature is not really being deprecated. 

It is best that breaking changes be overt.

**Proposal(s):**
1. A feature can be deprecated by any major or minor version. Add newly deprecated features to the Change Log.
1. A feature shall only be removed by a major version. Add newly removed features to the Change Log.
1. Make a general statement about deprecation (possibly in the Compliance section) and say how deprecated features are marked.
1. Mark the deprecated features thus.
1. Rather than quietly change the behavior of an existing feature, consider deprecating it and replacing it with something with a different name/syntax.
1. Any breaking change shall only be made by a major version.

### PR DESCRIPTION
**Pull request by kriswest**
resolves #531, resolves #532

Adds details of FDC3's versioning and deprecation policies to the complinace page in the website and adds a reference and current version number to the intro page.

### CODE REVIEW

**Review by RexJaeschke — APPROVED** [review]
Looks good to me!

File: docs/fdc3-compliance.md [hunk 1]
```diff
@@ -47,10 +47,16 @@ For application providers FDC3 compliance requires:
...
+
+## Deprecation Policy
+Over time, it is not uncommon for certain things in a standard to be marked for removal in a future version, possibly being replaced by an alternative. That is, they are deprecated. Often, they are retained in the standard because of their widespread use, but their use in new projects is discouraged.
+
+FDC3 adopts the following deprecation policy:
+
+1. A feature can be deprecated by any major or minor version. Newly deprecated features will be described in the [Changelog](https://github.com/finos/FDC3/blob/master/CHANGELOG.md).
+2. A feature shall only be removed by a major version. Newly removed features will be described in the [Changelog](https://github.com/finos/FDC3/blob/master/CHANGELOG.md).
+3. Deprecated features are clearly marked with an `@deprecated` tag and comment in both the documentation and jsDocs applied to the Typescript sources.
```
Comment (greyseer256) [inline docs/fdc3-compliance.md:60]: Typescript -> TypeScript

File: docs/fdc3-compliance.md [hunk 2]
```diff
@@ -47,10 +47,16 @@ For application providers FDC3 compliance requires:
...
+4. Where possible, changes to the behavior of an existing feature should be avoided; consider deprecating it and replacing it with something with a different name/syntax.
+5. Breaking change should only be made in a major version of the Standard.
```
Comment (greyseer256) [inline docs/fdc3-compliance.md:62]: Breaking change -> Breaking changes

File: website/README.md [hunk 3]
```diff
@@ -62,8 +62,9 @@ Since FDC3 is a schema project, we don't follow semver, which is meant for libra
 
 ## Create a new version
 
-Since the website also uses some generated and copied static files (like schemas), extra tasks need to be performed as part of creating
-a new version.
+Before creating a version, please make sure docs/fdc3-intro.md has been updated such the version number in the title and versioning section are correct for the version you are creating, as these have to be manually set.
```
Comment (greyseer256) [inline website/README.md:65]: such the -> such that the

**Review by greyseer256 — APPROVED** [review]
Minor comments, LGTM

File: docs/fdc3-compliance.md (hunk 1 above)
Comment (kriswest) [inline docs/fdc3-compliance.md:60]:
```suggestion
3. Deprecated features are clearly marked with an `@deprecated` tag and comment in both the documentation and jsDocs applied to the TypeScript sources.
```

File: docs/fdc3-compliance.md (hunk 2 above)
Comment (kriswest) [inline docs/fdc3-compliance.md:62]:
```suggestion
5. Breaking changes should only be made in a major version of the Standard.
```

File: website/README.md (hunk 3 above)
Comment (kriswest) [inline website/README.md:65]:
```suggestion
Before creating a version, please make sure docs/fdc3-intro.md has been updated such that the version number in the title and versioning section are correct for the version you are creating, as these have to be manually set.
```

**Review by rikoe — APPROVED** [review]

**Pull request merged**